docs/source/frameworks/tensorflow.rst

```diff
@@ -1,52 +1,244 @@
 ==========
-Tensorflow
+TensorFlow
 ==========
 
-Here's an example of serving Tensorflow models with BentoML:
+TensorFlow is an open source machine learning library focusing on training on inference of deep neural networks. BentoML provides native support for 
+serving and deploying models trained from TensorFlow.
 
-.. code:: python
+Preface
+-------
 
-    import bentoml
-    import tensorflow as tf
+Consider using BentoML :ref:`frameworks/keras:Keras` framework if working with a Keras model. If continuing with TensorFlow, make sure the Keras model 
+is decorated with :code:`tf.function`.
 
-    class NativeModel(tf.Module):
+.. note::
+
+    - Keras is not optimized for inference in production. There are reports about memory leaks during serving. :code:`bentoml.keras` has same concerns since it rely on the the Keras APIs.
+    - Inference with :code:`bentoml.tensorflow` is about twice faster than :code:`bentoml.keras`.
+    - :code:`bentoml.keras` performs input casting similar to the original Keras model for better debugging experiences.
```

Comment:
Should this be a part of the Keras guide instead?

Reply:
Do / should we have a separate Keras guide?

Reply:
Discussed with @sauyon, we think Keras doc should have a section for recommending converting Keras model to TensorFlow for better performance. In this section of the TF doc, we can advise not to use Keras in production and link to the conversion section of the Keras doc.

Reply:
@sauyon We have